### ISSUE TITLE
Reduce selection boxes for non-fully-grown plants

### ISSUE THREAD

**Issue by o11c** [FaceDeer/dfcaverns#9]
This could be done either as (current_stage / max_stage), or as a small fraction for *all* incomplete plants.

Among other things, this would provide a clear distinction between fully-grown and still-growing plants (for those that don't already have a color difference), preventing the latter from accidentally being harvested.

Sometimes it is possible to accidentally place a seed on top of a plant. When you then dig the seed, it falls to the ground and can't be picked up due to the plant's selection box.

(Mostly unrelated, but is there anything to be done about how `unified_inventory` shuffles the recipes?)

**Comment by FaceDeer** [comment]
Good idea, I'll look into fiddling with the selection boxes when I resume my renovations of this mod. I've been meaning to take a look at how other farming mods handle plant growth to make sure I wasn't doing something too weird in DFCaverns, I remember back at the beginning of this all having to roll my own farming code because everything in the default farming mod assumed that *light* was needed instead of *dark*.

What to you mean by unified_inventory shuffling the recipes? Something to do with how I used the dfcaverns:cookable group to make lots and lots of combinations of food items cookable?

**Comment by o11c** [comment]
> I remember back at the beginning of this all having to roll my own farming code because everything in the default farming mod assumed that light was needed instead of dark.

The one user-facing difference I see is that everything produces a *lot* more. Though there is huge variation between types of food (cave wheat relatively sucks, sweet pods overproduce if you also farm fuel, quarry bushes overproduce regardless, syrup is 3x more expensive for no better food)

> What to you mean by unified_inventory shuffling the recipes? Something to do with how I used the dfcaverns:cookable group to make lots and lots of combinations of food items cookable?

It's not just unified_inventory (though I noticed it there when I was trying to figure *out* all the recipes, and it lists them in random order), I hadn't tried it at the time ...

If you mix 4 different cookables, it's arbitrary-but-fixed (presumably by hash iteration order) which ones gets produced.

This could be fixed by using shaped recipes rather than shapeless (and always using the first/last ingredient). Most people are already going to only 1 or 2 shapes anyway.

There are more-complicated approaches that might be "better" (2-3 conceptual groups, perhaps flourlike and seasoninglike (maybe not a group, just the fixed item), maybe also seedlike). Notably, it doesn't make sense lorewise that quarry bush leaves can be mixed with themselves.

**Comment by FaceDeer** [comment]
The update I just rolled out to contentDB should do away with all the arbitrary recipes, they've all got specific ingredients now.
